File: C++/URI 1787.cpp
#include <stdio.h>
#include <vector>
#include <math.h>
#include <string>

using namespace std;

bool powOfTwo(int v){
    if (v == 1) return false;
    return v && !(v & (v - 1));
}

int main()
{
    int n, rita, uilton, ingred, u, r, i;
    string res;

    while (scanf("%d", &n) && n != 0)
    {
        rita = 0, uilton = 0, ingred = 0;
        for (size_t j = 0; j < n; j++)
        {
            scanf("%d %d %d", &u, &r, &i);
            if (powOfTwo(u)){
                uilton += 1;
                if (u > r && u > i)
                    uilton += 1;
            }
            if (powOfTwo(r)){
                rita += 1;
                if (r > u && r > i)
                    rita += 1;
            }
            if (powOfTwo(i)){
                ingred += 1;
                if (i > u && i > r)
                    ingred += 1;
            }
        }

        if (uilton > rita && uilton > ingred) res = "Uilton";
        else if (rita > uilton && rita > ingred) res = "Rita";
        else if (ingred > rita && ingred > uilton) res = "Ingred";
        else res = "URI";

        printf("%s\n", res.c_str());        

    }
    

    return 0;
}

File: Python/URI 1787.py
import math

while True:
    n = int(input())

    if (n == 0): break

    rita, uilton, ingred = 0, 0, 0
#   3       2       2
    for j in range(n):
        u, r, i = map(int, input().split())

        if ((math.log(u) / math.log(2)) % 2 == 0 or u == 2):
            uilton += 1
            if u > r and u > i:
                uilton += 1
        if ((math.log(r) / math.log(2)) % 2 == 0 or r == 2):
            rita += 1
            if r > u and r > i:
                rita += 1
        if ((math.log(i) / math.log(2)) % 2 == 0 or i == 2):
            ingred += 1
            if i > r and i > u:
                ingred += 1

    if uilton > rita and uilton > ingred: res = "Uilton"
    elif rita > uilton and rita > ingred: res = "Rita"
    elif ingred > rita and ingred > uilton: res = "Ingred"
    else: res = "URI"

    print(res)
